### DISCUSSION TITLE
How to figure out why package is not hoisted in yarn3 monorepo?

### DISCUSSION THREAD

**Discussion by AndriiTsarenko** [yarnpkg/berry#4472] [Q&A]
I'm using `yarn 3` monorepo and for some reason, the `react-dom` package is not hoisted properly: The same version of `react-dom` is duplicated across packages.
```
qnm react-dom
react-dom 18.1.0 ↰ 18 hours ago
├── 16.14.0 ⇡ 1 year ago
└─┬ package-a
  └── 16.14.0 ⇡ 1 year ago
```
I have the next resolutions in root `package.json`
```
  "resolutions": {
    "react": "^16.14.0",
    "react-dom": "^16.14.0",
    ...
  }
```
I read that package could be prevented from hoisting because of wrong peer dependencies, but seems that everything is a correct and hoisted version of `react` is meeting the peer dependency requirement of react-dom:
```
npm info react-dom@16.14.0 peerDependencies
{ react: '^16.14.0' }
```
Also, I've trying to do yarn dedupe and it's not helping.

Can you suggest any tool or approach how to debug why a package is not hoisted, with yarn 3?

Asked this question on [StackOverflow ](https://stackoverflow.com/questions/72288345/how-to-figure-out-why-package-is-not-hoisted-in-yarn3-monorepo) also.

**Comment by larixer** [comment]
Based on a discussion on [StackOverflow](https://stackoverflow.com/questions/72288345/how-to-figure-out-why-package-is-not-hoisted-in-yarn3-monorepo) the hoisting, in this case, was prevented due to Yarn's core auto-adding of peer dependencies on `@types/react` and `@types/react-dom`. Ideally, auto-adding of `@types/` peer dependencies should be reconsidered because it is unexpected and it actually harms `node_modules` installs.
